usr.sbin/inetd/inetd.c

```diff
@@ -2761,25 +2752,29 @@ rl_process(struct servtab *sep, int ctrl)
 
 		if (node != NULL) {
 			DPRINTF(
-			    "This service has a se_ip_max of %zu and an ip_count of %zu",
+			    "This service has se_ip_max %zu and ip_count %zu",
 			    sep->se_ip_max, node->count);
-			if (node->count == sep->se_ip_max) {
+			if (node->count >= sep->se_ip_max) {
 				if (!istimevalid) {
 					(void) gettimeofday(&now, NULL);
 				}
 			
 				if (now.tv_sec - sep->se_time.tv_sec > CNT_INTVL) {
 					state = DESTROY_IP_LIST;
 				} else {
-					syslog(LOG_ERR,
-                        		    "%s/%s max ip spawn rate (%zu in " 
-					    TOSTRING(CNT_INTVL) " seconds) for "
-					    "%." TOSTRING(NI_MAXHOST) "s "
-                        		    "exceeded; service not started.",
-                        		    sep->se_service,
-                        		    sep->se_proto,
-                        		    sep->se_ip_max,
-					    node->address);
+					if (debug) {
+						/* Only log during debug to prevent
+						DoS attack writing to system log */
+						syslog(LOG_ERR,
+                        		   	    "%s/%s max ip spawn rate (%zu in " 
+					   	    TOSTRING(CNT_INTVL) " seconds) for "
+					   	    "%." TOSTRING(NI_MAXHOST) "s "
+                        		   	    "exceeded; service not started.",
+                        		   	    sep->se_service,
+                        		   	    sep->se_proto,
+                        		   	    sep->se_ip_max,
+					   	    node->address);
+					}
```

Comment:
Since this IP rate limiting isn't perfect, and every incoming packet and connection is processed, this could lead to a lot of syslog garbage if there was a DoS attack from multiple IPs. Don't want any chance of that.